### ISSUE TITLE
child deletion security.

### ISSUE THREAD

**Issue by burdettadam** [Picolab/wrangler#108]
when a child pico is deleted there is a long chain of events that end with a scheduled event. this scheduled event is what removes the actual pico. 
It is possible for a malicious event to by pass the event chain and remove a single pico without thought about the child's children picos. This allows a rogue grand child to exist that is un-reachable by wrangler in traditional ways. 
the classic engine used special event domains that only "root" process could raise to prevent this.
An alternate solution could be creating and storing a uuid in an entity variable, sending the uuid as an attribute in the scheduled event. Then in the rule which is removing the pico, checking that the provided uuid is a valid stored uuid. This would increase the difficulty of creating rogue picos.

**Comment by 0joshuaolson1** [comment]
Of course, uuid's aren't designed for this kind of cryptographic purpose.
